### ISSUE TITLE
Customize type attribute of script

### ISSUE THREAD

**Issue by chenxsan** [jantimon/html-webpack-plugin#1663]
**Is your feature request related to a problem? Please describe.**

As of [5.41.0](https://github.com/webpack/webpack/releases/tag/v5.41.0), webpack will output `.mjs` by default when experimental ECMAScript modules support is enabled. You can see a demo [here](https://stackblitz.com/edit/node-axut71?file=webpack.config.js).

I'm wondering if `html-webpack-plugin` can support customizing the `type` attribute of `script` element:

```js
<script type='module' src=''></script>
```

**Describe the solution you'd like**

Add `type='module'` to `script` automatically when users enable experimental ESM support in `webpack.config.js`.

```js
module.exports = {
  experiments: {
    outputModule: true,
  },
};
```

**Describe alternatives you've considered**

Add a `type` option so users can customize as they wish.

**Additional context**

**Comment by jantimon** [comment]
Good idea - we already have the `scriptLoading` option which controls where and how tags are injected.

What about adding a new option here?

**Comment by chenxsan** [comment]
@jantimon Yeah, a new option would be suitable.

**Comment by jantimon** [comment]
I guess it could be `scriptLoading: 'module'`

**Comment by chenxsan** [comment]
Modules are [deferred automatically](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#other_differences_between_modules_and_standard_scripts), so there's no need to set one.

**Comment by hronro** [comment]
not stale
